app/models/collection.rb

```diff
@@ -335,6 +335,29 @@ def all_approved_bookmarks_count
     count
   end
 
+  # Return the count of all bookmarkable items (works, series, external works)
+  # that are in this collection (or any of its children) and visible to
+  # the current user. Excludes bookmarks of deleted works/series.
+  def all_bookmarked_items_count
+    # The set of all bookmarks in this collection and its children.
+    # Note that "approved_by_collection" forces the bookmarks to be approved
+    # both by the collection AND by the user.
+    bookmarks = Bookmark.is_public.joins(:collection_items).
+      merge(CollectionItem.approved_by_collection).
```

Comment:
Since Ruby inexplicably doesn't approve of start of line dots to indicate that the next line is a continuation of the previous one (sorry, Java habits coming to the fore), I second at least aligning the chained methods with the start of the chain to make it clearer this is still the chain and not a separate statement with accidental indentation.